cmd/kube-dns/SPEC.md

```diff
@@ -0,0 +1,140 @@
+    - Answer Format:
+      - `<hostname>.<service>.<ns>.svc.<zone>. <ttl> IN A <endpoint-ip>`
+    - Answer Example:
+      - `my-pet.my-service.default.svc.cluster.local. 4 IN A 10.3.0.100`
+- **Question:** `<a>-<b>-<c>-<d>.<ns>.pod.<zone>. IN A`
+  - If a Pod exists with IP address `<a>.<b>.<c>.<d>`, the result will be an A record with the specified name and the IP address `<a>.<b>.<c>.<d>`.
+    - Answer Format:
+      - `<a>-<b>-<c>-<d>.<ns>.pod.<zone>. <ttl> IN A <a>.<b>.<c>.<d>`
+    - Answer Example:
+      - `10-11-12-13.default.pod.cluster.local. 4 IN A 10.11.12.13`
+  - If no such Pod exists, the result will be `NXDOMAIN`.
+- **Question:** `_<port>._<proto>.<service>.<ns>.svc.<zone>. IN SRV`
+  - If no Service named `<service>` exists in `<ns>`, the result will be `NXDOMAIN`.
+  - If a Service named `<service>` exists in `<ns>` with a ClusterIP, and the service has a port named `<port>` using protocol `<proto>`, the result in the Answer section will be a single SRV record with the port number of the selected port and the name of the A record for the service (`<service>.<ns>.svc.<zone>.`). Unnamed ports will not receive an SRV record. The priority and weight returned are not prescribed by this specification. The Additional section may contain the named A record.
+    - Answer Format:
+       - `_<port>._<proto>.<service>.<ns>.svc.<zone>. <ttl> IN SRV <weight> <priority> <port-number> <service>.<ns>.svc.<zone>.`
+    - Answer Example:
+       - `_https._tcp.kubernetes.default.svc.cluster.local. 30 IN SRV 10 100 443 kubernetes.default.svc.cluster.local.`
+  - If a Service named `<service>` exists in `<ns>` without a ClusterIP, and the Service has no ready endpoints, the result will be an empty Answer section.
+  - If a Service named `<service>` exists in `<ns>` without a ClusterIP and with _ready_ endpoints, the result in the Answer section will be one SRV record for each _ready_ endpoint. Each record will contain the specific port number for the selected port and the name of an A record `<hostname>.<service>.<ns>.svc.<zone>` where `<hostname>` is the endpoint _hostname_. As described above, this name will resolve to an A record with the IP address of the specific _ready_ endpoint (which may be included in the Additional section). The priority and weight are not prescribed by this specification.
+    - Answer Format:
+      - `_<port>._<proto>.<service>.<ns>.svc.<zone>. <ttl> IN SRV <weight> <priority> <port-number> <hostname>.<service>.<ns>.svc.<zone>.`
+    - Answer Example:
+```
+      _https._tcp.headless.default.svc.cluster.local. 4 IN SRV 10 100 443 my-pet-1.headless.default.svc.cluster.local.
+      _https._tcp.headless.default.svc.cluster.local. 4 IN SRV 10 100 443 my-pet-2.headless.default.svc.cluster.local.
+      _https._tcp.headless.default.svc.cluster.local. 4 IN SRV 10 100 443 438934893.headless.default.svc.cluster.local.
+```
+- **Question:** `<service>.<ns>.svc.<zone>. IN SRV`
```

Comment:
I'm not sure about this set of records, as it doesn't follow the SRV RFC name format of _service._protocol. It might be a mistake. I would prefer to not have this if it is not used. I am going to ask the community to see if people are using it.